### PR TITLE
Documention: UML diagrams

### PR DESCRIPTION
**Pull request by t-h2o**
# [Preview available](https://t-h2o.github.io/webserv/)

### Documentation

* created resources section
* update titles
* updated SVG with docker MacOS
* UML diagrams
  * state diagram of a webserver
  * component diagram of a webserver
  * activity diagram of a webserver
  * sequence diagram of a webserver
  * usecase diagram of a webserver
* remove big-box documentation

### Makefile

* new rule for generating html with docker
* created DOCU and INDEX variables
* created ADOC variable

### Others

* Wikipedia: [UML](https://en.wikipedia.org/wiki/Unified_Modeling_Language)
* Docker image: [asciidoctor/docker-asciidoctor](https://hub.docker.com/r/asciidoctor/docker-asciidoctor)

### CODE REVIEW

**Review by RobinBurri — APPROVED** [review]
nice diagrams. Rien à redire.

File: Makefile [hunk 1]
```diff
@@ -6,6 +6,7 @@ CXX		=	g++
 AR		=	ar rs 2> /dev/null
 DB		=	lldb
 MAKE	=	make --no-print-directory 1> /dev/null
+ADOC	=	asciidoctor --require=asciidoctor-diagram
```
[the remark this answers is not in the document]
Comment (t-h2o) [reply Makefile:9]: [What is Asciidoctor?](https://docs.asciidoctor.org/asciidoctor/latest/#what-is-asciidoctor)

File: Makefile [hunk 2]
```diff
@@ -76,6 +76,10 @@ doc:
 	@printf "$(YELLOW)Generating documentations..$(DEFAULT)\n"
 	@$(ADOC) $(DOCU) -o $(INDEX)
 
+docdocker:
+	@printf "$(YELLOW)launch the asciidoctor/docker-asciidoctor docker image..$(DEFAULT)\n"
+	@docker run --rm -v $(shell pwd):/documents/ asciidoctor/docker-asciidoctor make doc
```
[the remark this answers is not in the document]
Comment (t-h2o) [reply Makefile:81]: With `man 1 docker-run`:

* `--rm`: automatically remove the container when it exits
* `-v`: mount the working directory into `/documents`
* launch `asciidoctor/docker-asciidoctor` image;
* `make doc` is executing into the docker image

**Review by 0xCAF3D0OD — APPROVED** [review]
c'est tout bon

**Pull request merged**